### ISSUE TITLE
Feature Request: Express logged foods in terms of serving sizes

### ISSUE THREAD

**Issue by BHSPitMonkey** [simonoppowa/OpenNutriTracker#34]
When I scan the barcode on a package of food (e.g. a bag of chips, a can of soup, etc.), it's counterintuitive that the default assumption is that I've eaten 100 grams of the product (rather than 1 serving, as described on the label).

Additionally, if I know I've had some multiple (like 1.5x or 2x) of servings, I'm seemingly expected as a user to open up a calculator app, read the weight off the package, divide that by the labeled "servings per container", and multiply that by the number of servings I've had to get the desired consumption amount in grams (and then switch back to the app, and enter that value).

In competing apps, the looked-up metadata includes the indicated serving size and I can easily just change the default (in servings) from "1" to some other amount (and the other values on the diary entry will scale accordingly). And, importantly, this multiplication is only associated with the diary entry and not the food definition itself (so I can easily have 2 servings today and 3 tomorrow).

Thanks for building this and releasing under the GPL! It's off to a promising start, and with some adjustments I could see it being a daily driver for myself and others.

**Comment by ossaleon** [comment]
I totally agree with @BHSPitMonkey. An easy implementation could be to add servings as a possible unit for the items that have a serving size, instead of just g/ml. I would be happy to contribute with this feature, I'm using the app since a couple of weeks and I totally love it, but this feature would be very useful.

**Comment by mblennegard** [comment]
This is much needed and as mentioned by @BHSPitMonkey also what is often expected when using a calorie tracker.

For me personally I mostly register by weight but some things, e.g. frozen items like burgers it is much easier to register by servings.

Either way, the current state in the app is a bit counter-intuitive as there actually is a serving size to be entered when creating a new meal item in the app, but then when adding said item to the diary the drop-down containing the unit (g, ml) cannot be clicked. From a UX perspective it seems reasonable to simply have this drop-down be enabled at all times.

**Comment by colin-bytheway** [comment]
I'm equally confused by the concept of "kcal per g/ml" and "kcal per 100 g/ml".
Being able to enter a product's Nutrition Facts and enter in what I see would be more user friendly. For example, 1serving size = 170g, 160 cals, 16g protein, etc. The fields and unit representations are confusing ("Meal size (oz/fl oz", "Base quantity (g/ml)", "kcal per100 g/ml", carbs per100 g/ml", etc.).

Sorry if this doesn't meet standard Contributing Guidelines. I couldn't find any and this is my first time contributing to anything!

<img width="1014" alt="Image" src="https://github.com/user-attachments/assets/8934daaa-d0f6-4ad0-bf8e-d0a95d3cebe4" />

**Comment by jordan-lee-code** [comment]
Hi @BHSPitMonkey — partial fix; confirming the remainder is still outstanding on `develop`.

Code location: there are two paths into the meal-detail bottom sheet, with inconsistent defaults:
- `lib/features/meal_detail/meal_detail_screen.dart:78-79` sets `_initialUnit = UnitDropdownItem.serving.toString()` if `meal.hasServingValues` — this branch correctly defaults to serving when available.
- `lib/features/meal_detail/presentation/bloc/meal_detail_bloc.dart:44` initializes the bloc state with `selectedUnit: UnitDropdownItem.gml.toString()` — this still defaults to g/ml unconditionally.

The screen-level branch helps for some entry paths but the bloc default trumps in others. Same fix as #158: branch `meal_detail_bloc.dart:44` on `event.product.servingSize != null`.

Labelled `enhancement`. Tracked alongside #158 (default-to-serving on barcode scan).

**Comment by jordan-lee-code** [comment]
Merged in #391. Thanks for filing — this is now on `develop` and will ship with the next release.

If you spot something still wrong on a fresh build, please reopen this issue or open a new one with the details and I'll have a look.
